### PR TITLE
Add `daysAgo` filter

### PR DESCRIPTION
**Pull request by frankieroberto**
This accepts an ISO date string and returns the number of days ago that it was.

### CODE REVIEW

File: lib/date.js [hunk 1]
```diff
@@ -48,6 +48,38 @@ function duration (string, number, unit) {
...
+ */
+function daysAgo (string) {
+  const timeZone = 'Europe/London'
+
+  string = normalize(string, '')
+  try {
+    const date = DateTime.fromISO(string, { zone: timeZone })
+    const today = DateTime.now().setZone(timeZone)
+
+    return Math.floor(today.diff(date, 'days').toObject().days)
```
Comment (frankieroberto) [inline lib/date.js]: I did start out by using the native Date objects, with `Date.parse()` and `Math.floor((today.getTime() - date.getTime()) / (1000 * 3600 * 24))` which worked fine, but decided to switch to the luxon objects for consistency with the other methods, and better error handling if an invalid date string is given.

File: docs/date.md [hunk 2]
```diff
@@ -55,6 +55,57 @@ To return a date from today’s date, pass the special word `"today"` (or `"now"
...
+
+{% if leftDaysAgo > 0 %}
+  They have left the organisation on
+  {{ data | isoDateFromDateInput("leavingDate") | govukDate }}.
+{% elif  leftDaysAgo == 0 %}
+  They are leaving the organisation today.
+{% else %}
+  They will be leaving the organisation on
+  {{ data | isoDateFromDateInput("leavingDate") | govukDate }}.
+{% end %}
```
Comment (paulrobertlloyd) [inline docs/date.md]:
```suggestion
{% if leftDaysAgo > 0 %}
  They left the organisation on
  {{ data | isoDateFromDateInput("leavingDate") | govukDate }}.
{% elif leftDaysAgo == 0 %}
  They are leaving the organisation today.
{% else %}
  They will be leaving the organisation on
  {{ data | isoDateFromDateInput("leavingDate") | govukDate }}.
{% end %}
```

**Review by paulrobertlloyd — APPROVED** [review]
Nice one. Just a few typos spotted.

**Comment by paulrobertlloyd** [comment]
Another minor thing; I’ve been keeping filters alphabetically ordered, both in docs and imports/exports and test order. Super pedantic, I know. So I think that means this filter should appear first in those places.

**Comment by paulrobertlloyd** [comment]
Once this is merged, shall we release as v1.1?

**Comment by frankieroberto** [comment]
> Another minor thing; I’ve been keeping filters alphabetically ordered, both in docs and imports/exports and test order. Super pedantic, I know. So I think that means this filter should appear first in those places.

@paulrobertlloyd Yeah that is pedantic! 😉  I've done it now though.

Longer-term, I wonder if it'd make sense to have a file-per-function, so something like `lib/date/daysAgo.js` and then similarly the tests could be `tests/data/daysAgo.mjs`?  That way we don't have to worry about the order? Might make it easier to locate the tests too?

**Comment by paulrobertlloyd** [comment]
> I wonder if it'd make sense to have a file-per-function

I did toy with that idea originally I think, but it seemed overkill. But as more filters are added, and especially those that are either a.) hard to categorise or b.) fall into multiple categories, that might make more sense. (Though might still want to categorise in the docs…)

**Pull request merged**